docs/src/test/scala/docs/http/scaladsl/HttpServerExampleSpec.scala

```diff
@@ -699,5 +699,49 @@ class HttpServerExampleSpec extends WordSpec with Matchers
     //#discard-close-connections
   }
 
+  "dynamic routing example" in compileOnlySpec {
+    //#dynamic-routing-example
+    import akka.actor.ActorSystem
+    import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+    import akka.http.scaladsl.server.Directives._
+    import akka.http.scaladsl.server.Route
+    import akka.stream.ActorMaterializer
+    import spray.json.DefaultJsonProtocol._
+    import spray.json._
+    import scala.collection.concurrent.TrieMap
+
+    case class MockDefinition(path: String, requests: Seq[JsObject], responses: Seq[JsObject])
+
+    implicit val format = jsonFormat3(MockDefinition)
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorMaterializer()
+
+    val state = TrieMap.empty[String, Map[JsObject, JsObject]]
 
+    // fixed route to update state
+    val fixedRoute: Route = post {
+      pathSingleSlash {
+        entity(as[MockDefinition]) { mock =>
+          val mapping = mock.requests.zip(mock.responses).toMap
+          state.put(mock.path, mapping)
```

Comment:
While `TrieMap` is indeed thread-safe, it might be a better example to replace the `Map`?